owncloudData/src/test/java/com/owncloud/android/data/capabilities/repository/OCCapabilityRepositoryTest.kt

```diff
@@ -41,95 +44,67 @@ class OCCapabilityRepositoryTest {
     @JvmField
     val instantExecutorRule = InstantTaskExecutorRule()
 
-    private val localCapabilitiesDataSource = mockk<LocalCapabilitiesDataSource>(relaxed = true)
-    private val remoteCapabilitiesDataSource = mockk<RemoteCapabilitiesDataSource>(relaxed = true)
-    private val appRegistryRepository = mockk<AppRegistryRepository>()
+    private val localCapabilitiesDataSource = mockk<LocalCapabilitiesDataSource>(relaxUnitFun = true)
+    private val remoteCapabilitiesDataSource = mockk<RemoteCapabilitiesDataSource>()
+    private val appRegistryRepository = mockk<AppRegistryRepository>(relaxUnitFun = true)
     private val ocCapabilityRepository: OCCapabilityRepository =
         OCCapabilityRepository(localCapabilitiesDataSource, remoteCapabilitiesDataSource, appRegistryRepository)
 
     @Test
-    fun refreshCapabilitiesFromNetworkOk() {
-        val capability = OC_CAPABILITY.copy(accountName = OC_ACCOUNT_NAME)
-
-        every { remoteCapabilitiesDataSource.getCapabilities(any()) } returns capability
-
-        ocCapabilityRepository.refreshCapabilitiesForAccount(OC_ACCOUNT_NAME)
-
-        verify(exactly = 1) {
-            remoteCapabilitiesDataSource.getCapabilities(OC_ACCOUNT_NAME)
-        }
+    fun `getCapabilitiesAsLiveData returns a Livedata of OCCapability`() {
+        val capabilitiesLiveData = MutableLiveData<OCCapability>()
 
-        verify(exactly = 1) {
-            localCapabilitiesDataSource.insertCapabilities(listOf(capability))
-        }
-    }
+        every {
+            localCapabilitiesDataSource.getCapabilitiesForAccountAsLiveData(OC_ACCOUNT_NAME)
+        } returns capabilitiesLiveData
 
-    @Test(expected = NoConnectionWithServerException::class)
-    fun refreshCapabilitiesFromNetworkNoConnection() {
+        val capabilitiesEmitted = ocCapabilityRepository.getCapabilitiesAsLiveData(OC_ACCOUNT_NAME).getLastEmittedValue()
 
-        every { remoteCapabilitiesDataSource.getCapabilities(any()) } throws NoConnectionWithServerException()
+        val capabilitiesToEmit = capabilitiesLiveData.getLastEmittedValue()
```

Comment:
The logical order of a test is first mocking/constructing the expected result and then obtaining the actual result, then comparing both

Reply:
Changed the order